Tools/TestWebKitAPI/Tests/WebKitCocoa/LoadAndDecodeImage.mm

```diff
@@ -156,4 +156,33 @@
     Util::run(&done);
 }
 
+TEST(WebKit, GetInformationFromImageData)
+{
+    RetainPtr webView = adoptNS([WKWebView new]);
+
+    done = false;
+    RetainPtr pngData = [NSData dataWithContentsOfURL:[NSBundle.test_resourcesBundle URLForResource:@"icon" withExtension:@"png"]];
+    [webView _getInformationFromImageData:pngData.get() completionHandler:^(BOOL canBeDecodedAsSVG, NSArray<NSValue *> *availableSizes, NSError *error) {
+        EXPECT_NULL(error);
+        EXPECT_FALSE(canBeDecodedAsSVG);
+        EXPECT_EQ(1u, availableSizes.count);
+        for (NSValue *size in availableSizes) {
+            EXPECT_EQ(215, [size sizeValue].width);
+            EXPECT_EQ(174, [size sizeValue].height);
+        }
+        done = true;
+    }];
+    Util::run(&done);
+
+    done = false;
+    RetainPtr svgData = [NSData dataWithContentsOfURL:[NSBundle.test_resourcesBundle URLForResource:@"AllAhem" withExtension:@"svg"]];
+    [webView _getInformationFromImageData:svgData.get() completionHandler:^(BOOL canBeDecodedAsSVG, NSArray<NSValue *> *availableSizes, NSError *error) {
```

Comment:
It seems like this is the first of two steps, the second of which would be actually getting the bitmap or svg.  Is that correct?

Reply:
Yes, we will have a different SPI for getting the actual image. This SPI is only about extracting information so that client can decide whether to proceed.